pkg/acquisition/modules/docker/docker.go

```diff
@@ -924,34 +1035,62 @@ func (d *DockerSource) TailContainer(ctx context.Context, container *ContainerCo
 		case <-readerTomb.Dying():
 			// This case is to handle temporarily losing the connection to the docker socket
 			// The only known case currently is when using docker-socket-proxy (and maybe a docker daemon restart)
-			d.logger.Debugf("readerTomb dying for container %s, removing it from runningContainerState", container.Name)
-			deleteChan <- container
-			// Also reset the Since to avoid re-reading logs
-			d.Config.Since = time.Now().UTC().Format(time.RFC3339)
-			d.containerLogsOptions.Since = d.Config.Since
-
-			return nil
+			container.logger.Debugf("readerTomb dying for container %s, connection lost", container.Name)
+			readerTomb.Kill(nil)
+			return fmt.Errorf("reader connection lost for container %s", container.Name)
 		}
 	}
 }
 
 func (d *DockerSource) TailService(ctx context.Context, service *ContainerConfig, outChan chan types.Event, deleteChan chan *ContainerConfig) error {
-	service.logger.Info("start tail")
+	service.logger.Info("start monitoring")
+
+	for {
+		select {
+		case <-service.t.Dying():
+			service.logger.Infof("tail stopped for service %s", service.Name)
+			return nil
+		default:
+		}
+
+		err := d.tailServiceAttempt(ctx, service, outChan)
+
+		if err == nil {
+			// Successful completion - service was stopped gracefully
+			return nil
+		}
+
+		// Check service health to determine if we should retry or give up
+		serviceHealthy := d.isServiceStillRunning(ctx, service)
+
+		if !serviceHealthy {
+			// Service was removed - don't retry, remove from monitoring
+			service.logger.Infof("service %s no longer exists, removing from monitoring: %v", service.Name, err)
+			deleteChan <- service
```

Comment:
im un decided if the container should self delete, since we get docker events now we already get this.

my fear is that it may cause a race condition like:

docker event die -> docker event alive -> container self delete